src/main/java/be/seeseemelk/mockbukkit/entity/FishHookMock.java

```diff
@@ -0,0 +1,214 @@
+package be.seeseemelk.mockbukkit.entity;
+
+import be.seeseemelk.mockbukkit.ServerMock;
+import be.seeseemelk.mockbukkit.UnimplementedOperationException;
+import net.kyori.adventure.text.Component;
+import org.apache.commons.lang.Validate;
+import org.bukkit.Location;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.FishHook;
+import org.bukkit.entity.SpawnCategory;
+import org.bukkit.util.Vector;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.UUID;
+
+public class FishHookMock extends ProjectileMock implements FishHook
+{
+
+	private int minWaitTime = 100;
+	private int maxWaitTime = 600;
+	private boolean applyLure = true;
+	private double biteChance = -1;
+	private Entity hookedEntity;
+	private HookState state = HookState.UNHOOKED;
+
+	public FishHookMock(@NotNull ServerMock server, @NotNull UUID uuid)
+	{
+		super(server, uuid);
+	}
+
+	@Override
+	public int getMinWaitTime()
+	{
+		return this.minWaitTime;
+	}
+
+	@Override
+	public void setMinWaitTime(int minWaitTime)
+	{
+		Validate.isTrue(minWaitTime >= 0 && minWaitTime <= this.getMaxWaitTime(), "The minimum wait time should be between 0 and the maximum wait time.");
+		this.minWaitTime = minWaitTime;
+	}
+
+	@Override
+	public int getMaxWaitTime()
+	{
+		return this.maxWaitTime;
+	}
+
+	@Override
+	public void setMaxWaitTime(int maxWaitTime)
+	{
+		Validate.isTrue(maxWaitTime >= 0 && maxWaitTime >= this.getMinWaitTime(), "The maximum wait time should be higher than or equal to 0 and the minimum wait time.");
+		this.maxWaitTime = maxWaitTime;
+	}
+
+	@Override
+	public boolean getApplyLure()
+	{
+		return this.applyLure;
+	}
+
+	@Override
+	public void setApplyLure(boolean applyLure)
+	{
+		this.applyLure = applyLure;
+	}
+
+	@Override
+	public double getBiteChance()
+	{
+		if (this.biteChance == -1)
+		{
+			// todo: Uncomment when WorkMock#isClearWeather is implemented.
+//			if (!getWorld().isClearWeather()) {
+//				return 1 / 300.0;
+//			}
```

Comment:
Let's implement that feature first so we can make this PR great again

Reply:
Implemented in #379 so just waiting on that.